opensearch_py_ml/ml_models/semantic_highlighter_model.py

```diff
@@ -0,0 +1,350 @@
+            Model ID to use, by default DEFAULT_MODEL_ID
+        model_name : str, optional
+            Name for the traced model file, by default None
+        save_json_folder_path : str, optional
+            Path to save config files, by default None
+        model_output_path : str, optional
+            Path to save the traced model, by default None
+        zip_file_name : str, optional
+            Name for the zip file, by default None
+        add_apache_license : bool, optional
+            Whether to add Apache license, by default True
+
+        Returns
+        -------
+        str
+            Path to the created zip file
+        """
+        if model_name is None:
+            model_name = str(model_id.split("/")[-1] + ".pt")
+
+        # Create output directories
+        os.makedirs(self.folder_path, exist_ok=True)
+        model_path = os.path.join(self.folder_path, model_name)
+
+        if save_json_folder_path is None:
+            save_json_folder_path = self.folder_path
+
+        if model_output_path is None:
+            model_output_path = self.folder_path
+
+        if zip_file_name is None:
+            zip_file_name = str(model_id.split("/")[-1] + ".zip")
+
+        zip_file_path = os.path.join(model_output_path, zip_file_name)
+
+        # Initialize and trace the model
+        model = TraceableBertTaggerForSentenceExtractionWithBackoff.from_pretrained(
+            model_id
+        )
+        tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
+        model = model.to("cpu")
+
+        # Save tokenizer files
+        tokenizer_path = os.path.join(self.folder_path, "tokenizer")
+        os.makedirs(tokenizer_path, exist_ok=True)
+        tokenizer.save_pretrained(tokenizer_path)
+        print(f"Tokenizer files saved to {tokenizer_path}")
+
+        # Trace the model
+        traced_model = torch.jit.trace(
+            model,
+            (
+                example_inputs["input_ids"],
+                example_inputs["attention_mask"],
+                example_inputs["token_type_ids"],
+                example_inputs["sentence_ids"],
+            ),
+        )
+
+        # Save the traced model
+        torch.jit.save(traced_model, model_path)
+        print(f"Model file saved to {model_path}")
+
+        # Create zip file with model, tokenizer, and config
+        with ZipFile(str(zip_file_path), "w") as zipObj:
+            # Add model file
+            zipObj.write(model_path, arcname=str(model_name))
+
+            # Add tokenizer files directly to root
+            for file in os.listdir(tokenizer_path):
+                file_path = os.path.join(tokenizer_path, file)
+                zipObj.write(file_path, arcname=file)
+
+        if add_apache_license:
+            super()._add_apache_license_to_model_zip_file(zip_file_path)
+
+        self.torch_script_zip_file_path = zip_file_path
+        print(f"Zip file saved to {zip_file_path}")
+        return zip_file_path
+
+    def save_as_onnx(
```

Comment:
What is the bottleneck for onnx? Have we tried to export it onnx?

Reply:
Support onnx format isn't in scope of p0 release, we can look into it later. cc: @greaa-aws